lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -767,9 +790,14 @@ impl ShardReplicaSet {
 
         locally_disabled.remove(&peer_id_to_remove);
 
+        self.notify_peer_failure_backoff
+            .write()
+            .reset(peer_id_to_remove);
+
         if active_peers.is_subset(&locally_disabled) {
             log::warn!("Resolving consensus/local state inconsistency");
```

Comment:
The idea is that if at some point we have *all* active peers "locally disabled" on a node, we treat is as a major consensus/local state desync and clear locally disabled peers list.

If all these nodes are truly dead, then following updates would fail, and we'll add them back to the locally disabled list. But if they're *not* dead (e.g., there was a network issue on this particular node, and it marked everything as dead), then we better clear the locally disabled list.